v2/managedsettling.go

```diff
@@ -27,9 +27,11 @@ func (m *ManagedSettler) Handle(ctx context.Context, settler MessageSettler, mes
 	}
 	if err := settler.CompleteMessage(ctx, message, nil); err != nil {
 		log(ctx, err)
+		m.options.OnFailure(ctx, message, err)
 		// if we fail to complete the message, we log the error and let the message lock expire.
 		// we cannot do more at this point.
 	}
+	m.options.OnComplete(ctx, message)
```

Comment:
do we want to have onComplete just call the complete settler by default?

Reply:
no, because that's a hook for the user to do stuff like logging. not to modify the behavior of the settling entirely. 
for that you can implement your own handler.